setup.py

```diff
@@ -124,13 +124,29 @@ def initialize_options(self) -> None:
     'nose'
 ]
 
+smart_open_dependencies = [
```

Comment:
This file, one of the 3 largest in the project, increased in size to 358 lines.  The total size of those files is now 3400 lines (target: 1086).  Is this file complex enough to refactor?